git.md

```diff
@@ -1,2 +1,139 @@
+to be focused; they are easier to review, and easier to merge because they
+offer fewer opportunities for conflicts.
+
+## 4. The work you share should be complete
+
+Each commit and feature should do what it promises, completely; and reverting
+it should undo what that commit or feature meant to do, also completely. If
+the goal of one commit requires changes in two files, that one commit should
+include changes to both files. Similarly, if the goal of a feature requires
+two kinds of changes, that feature should likely include two commits.
+
+## 5. Explain how to test the effect of the changes you made
+
+Testing your code is important, particularly when sharing it with
+others. Automated tests are the best but not the only way to test the effects
+of the changes you made. However you did it, explain how you ensured that the
+code behaves as you expect, and how someone else may reproduce your results.
+
+## 6. Structure each commit message like an email
+
+Like an email, commit messages have a subject and an optional body. The subject
+should start with a line summarizing your changes (aim for 50 characters or
+less). Use the imperative, present tense -- for example, "Fix typo",
+not "Fixed typo" or "Fixes typo". The message body, if present,
+should follow a blank line; wrap around 70 characters; and explain the
+motivation for the change ("why", not "what" or "how").
+
+See also:
+
+* [Example of a commit message](https://github.com/2DegreesInvesting/resources/issues/74).
+* [Commit best practices](https://r-pkgs.org/git.html#commit-best-practices).
+
+## 7. Agree on a workflow
+
+Choose a workflow based on your team's needs and preferences. However you
+choose to work, just make sure to agree on a common workflow that everyone
+follows:
+
+GitHub workflow:
+
+This workflow is 2DII's default. If a project uses a different
+workflow, document prominently (e.g. in README) what that workflow is
+([example](https://github.com/github/gitignore#contributing-workflow)).
+
+* Uses a central repository. 
+* Uses a single long-lived branch: "main"; it never contains broken code.
+* Developers work on "feature" branches (never on the "main" branch), share
+them via pull requests, and peers review the changes before they become a
+part of the main codebase.
+* A new release is created by tagging a particular commit on the main branch.
+
+Gitflow workflow:
+
+* Like the GitHub workflow but uses two long-lived branches: "main"
+and "develop". The main branch stores the official release history,
+and the develop branch serves as an integration branch for features.
+
+A team may agree on custom conventions beyond the "vanilla" workflows listed
+above.  For example, you may agree to prefix a commit-title with "WIP",
+"FIXME", "Amend: " or to whatever to communicate with your team succinctly
+via a vocabulary you all share.
+
+See also: 
+
+* [Comparing workflows](https://www.atlassian.com/git/tutorials/comparing-workflows).
+
```

Comment:
I think that we should also include a section on the value of local and global `gitignore`s. Local gitignores should call out what should be ignored for that project (for example, data files or rendered outputs), while global gitignores should be the files that each developer should ignore in _every_ project (`.DS_Store`, vim swap files, anything that is specific to my particular machine or text editor configuration)

Reply:
Thanks for pointing that out. 

@jdhoffa also pushed for .gitignore files and now the sectoin _2. Keep commits focused_ inlcudes "You can avoid sharing unwanted changes via .gitignore." 

Your suggestion would fit nicely here:

>  You can avoid sharing unwanted changes via .gitignore. Local gitignores should call out what should be ignored for that project (for example, data files or rendered outputs), while global gitignores should be the files that each developer should ignore in every project (.DS_Store, vim swap files, anything that is specific to my particular machine or text editor configuration)

To keep this document short, this document focuses only on _what_ to do and _why_; discussions about _how_ to do it are referenced via links ([.gitignore templates](.gitignore templates)). @jacobvjk and I discussed this restriction and figured we need to draw the boundary somewhere. But I realize this decision paves the road for documenting other themes so we likely need to engage @MonikaFu and the leaders of other themes like you.

What do you think? Is this a reasonable trade-off? Should we continue in the tab "Discussions"?

Reply:
I fall pretty solidly into the mindset that each person working with the repo is responsible for keeping their contributions clean, not for cleaning up after other people. If editor-specific files are added to the `.gitignore`, then it's easy for that to grow to include a lot of unnecessary filetypes, especially if they are there after people move along. And if we put user-specific files into the local ignore, then we have to put it in there for _every_ repo, which also sounds like a worse option than coaching people to build their own gloabl `.gitignore` and move it from machine to machine with them, like they would any other configuration files.

Reply:
> I fall pretty solidly into the mindset that each person working with the repo is responsible for keeping their contributions clean, not for cleaning up after other people.

Then we don't need `.gitignore` (global or repo-specific) at all... everyone should just do their commits properly and not include stuff that they shouldn't. Obviously, I'm being a bit sarcastic.

> If editor-specific files are added to the .gitignore, then it's easy for that to grow to include a lot of unnecessary filetypes

Hypothetically I agree with this, but I'm willing to bet that if we searched the entire git history of every repo 2dii has ever had that we would find no more than 20 file extensions or filenames of hidden files or other stuff that one would be likely to put in `.gitignore`

Reply:
I don't know of any links that specifically address the debate of using global .gitignore versus repo specific .gitignore versus not using either, but I agree that whether or not and how one uses .gitignore is an implementation detail. The guideline is: do not commit unnecessary files into the repo. The advice is: consider using global and/or repo .gitignore files to help you not do that.

https://docs.github.com/en/github/using-git/ignoring-files
https://gist.github.com/octocat/9257657
https://github.com/github/gitignore/blob/master/R.gitignore

Reply:
Note to myself: Maybe link the word .gitignore to https://git-scm.com/docs/gitignore